src/ImageSharp.Formats.Jpeg/JpegEncoderCore.cs

```diff
@@ -168,18 +180,17 @@ public void Encode<TColor>(Image<TColor> image, Stream stream, int quality, Jpeg
                 throw new ImageFormatException($"Image is too large to encode at {image.Width}x{image.Height}.");
             }
 
-            this.outputStream = stream;
-            this.subsample = sample;
-
-            if (quality < 1)
+            // Ensure that quality can be set but has a fallback.
+            int quality = this.options.Quality > 0 ? this.options.Quality : image.MetaData.Quality;
+            if (quality == 0)
             {
-                quality = 1;
+                quality = 75;
             }
 
-            if (quality > 100)
-            {
-                quality = 100;
-            }
+            quality = quality.Clamp(1, 100);
+
+            this.outputStream = stream;
+            this.subsample = this.options.Subsample ?? (quality >= 91 ? JpegSubsample.Ratio444 : JpegSubsample.Ratio420);
```

Comment:
I think we have that marked as 80 and below for 420 in the quality XML comments, 91 is the correct value for switching so we should say 90.